cmd/geoipupdate/main.go

```diff
@@ -52,27 +58,60 @@ func main() {
 
 	client := geoipupdate.NewClient(config)
 
-	if err = run(client, config); err != nil {
+	downloadFunc := func(editionID string) error {
+		return download(client, config, editionID)
+	}
+
+	if err = run(config, downloadFunc); err != nil {
 		fatalLogger("error retrieving updates", err)
 	}
 }
 
-func run(client *http.Client, config *geoipupdate.Config) error {
-	dbReader := database.NewHTTPDatabaseReader(client, config)
-
+// run concurrently downloads GeoIP databases using the provided config.
+// `config.Parallelism` limits the number of concurrent downloads that can be
+// executed and setting it to 1 would just mean databases would download
+// sequencially.
+func run(
+	config *geoipupdate.Config,
+	downloadFunc func(string) error,
+) error {
+	g := new(errgroup.Group)
```

Comment:
I was going to suggest using `errgroup.WithContext` here, but we don't seem to be passing the context to anything in this code. I guess that is an improvement for a future major version.